.brigade/brigade.ts

```diff
@@ -0,0 +1,292 @@
+
+const buildBrigadierJobName = "build-brigadier"
+const buildBrigadierJob = (event: Event) => {
+  return new MakeTargetJob(buildBrigadierJobName, jsImg, event)
+}
+jobs[buildBrigadierJobName] = buildBrigadierJob
+
+const publishBrigadierJobName = "publish-brigadier"
+const publishBrigadierJob = (event: Event) => {
+  return new MakeTargetJob(publishBrigadierJobName, jsImg, event, {
+    "NPM_TOKEN": event.project.secrets.npmToken
+  })
+}
+jobs[publishBrigadierJobName] = publishBrigadierJob
+
+const buildCLIJobName = "build-cli"
+const buildCLIJob = (event: Event) => {
+  return new MakeTargetJob(buildCLIJobName, goImg, event)
+}
+jobs[buildCLIJobName] = buildCLIJob
+
+const publishCLIJobName = "publish-cli"
+const publishCLIJob = (event: Event) => {
+  return new MakeTargetJob(publishCLIJobName, goImg, event, {
+    "GITHUB_ORG": event.project.secrets.githubOrg,
+    "GITHUB_REPO": event.project.secrets.githubRepo,
+    "GITHUB_TOKEN": event.project.secrets.githubToken
+  })
+}
+jobs[publishCLIJobName] = publishCLIJob
+
+const publishChartJobName = "publish-chart"
+const publishChartJob = (event: Event) => {
+  return new MakeTargetJob(publishChartJobName, helmImg, event, {
+    "HELM_REGISTRY": event.project.secrets.helmRegistry || "ghcr.io",
+    "HELM_ORG": event.project.secrets.helmOrg,
+    "HELM_USERNAME": event.project.secrets.helmUsername,
+    "HELM_PASSWORD": event.project.secrets.helmPassword
+  })
+}
+jobs[publishChartJobName] = publishChartJob
+
+// Run the entire suite of tests WITHOUT publishing anything initially. If
+// EVERYTHING passes AND this was a push (merge, presumably) to the v2 branch,
+// then run jobs to publish "edge" images.
+async function runSuite(event: Event): Promise<void> {
+  await new SerialGroup(
+    new ConcurrentGroup( // Basic tests
+      testUnitGoJob(event),
+      lintGoJob(event),
+      testUnitJSJob(event),
+      lintJSJob(event),
+      yarnAuditJob(event),
+      lintChartJob(event),
+      validateSchemasJob(event)
+    ),
+    new ConcurrentGroup( // Build everything
+      buildAPIServerJob(event),
+      buildGitInitializerJob(event),
+      buildLoggerLinuxJob(event),
+      buildObserverJob(event),
+      buildSchedulerJob(event),
+      buildWorkerJob(event),
+      buildBrigadierJob(event),  
+      buildCLIJob(event)
+    )
+  ).run()
+  if (event.worker?.git?.ref == "/refs/heads/v2") {
+    // Push "edge" images.
+    //
+    // npm packages MUST be semantically versioned, so we DON'T publish an
+    // edge brigadier package.
+    //
+    // To keep our github released page tidy, we're also not publishing "edge"
+    // CLI binaries.
+    await new ConcurrentGroup(
+      pushAPIServerJob(event),
+      pushGitInitializerJob(event),
+      pushLoggerLinuxJob(event),
+      pushObserverJob(event),
+      pushSchedulerJob(event),
+      pushWorkerJob(event),
+    ).run()
+  }
+}
+
+// Either of these events should initiate execution of the entire test suite.
```

Comment:
Can we also add handlers for `issue_comment:created` and `issue_comment:edited` for `/brig run` and `/brig check` support?

Reply:
There's no need. The new gateway does all of that. In the same way that it automatically talks back to github to request a check_suite when an eligible PR is opened or synchronized, it now automatically talks back to github to request a check_suite for eligible issue comment events.

Reply:
So what's nice is users of that gateway, in most cases, will just want to subscribe to various check events, without concern for _how_ or why the check was requested.

Reply:
Ahh, nice!  I could've/should've realized that from reviewing the v2 GH gateway 😄 

Super slick.

Reply:
No worries. It was a blink and you miss it sorta thing.